iib/web/models.py

```diff
@@ -1943,3 +1949,98 @@ def get_mutable_keys(self) -> Set[str]:
         return rv
+
+
+class RequestFbcOperations(Request, RequestIndexImageMixin):
+    """FBC operation build request."""
+
+    __tablename__ = 'request_fbc_operations'
+
+    id = db.Column(db.Integer, db.ForeignKey('request.id'), autoincrement=False, primary_key=True)
+
+    fbc_fragment_id = db.Column(db.Integer, db.ForeignKey('image.id'))
+    fbc_fragment_resolved_id = db.Column(db.Integer, db.ForeignKey('image.id'))
+    fbc_fragment = db.relationship('Image', foreign_keys=[fbc_fragment_id], uselist=False)
+    fbc_fragment_resolved = db.relationship(
+        'Image', foreign_keys=[fbc_fragment_resolved_id], uselist=False
+    )
+    build_tags = None
+
+    __mapper_args__ = {
+        'polymorphic_identity': RequestTypeMapping.__members__['fbc_operations'].value
+    }
+
+    @classmethod
+    def from_json(  # type: ignore[override] # noqa: F821
+        cls,
+        kwargs: FbcOperationRequestPayload,
+    ):
+        """
+        Handle JSON requests for the fbc-operations API endpoint.
+
+        :param dict kwargs: the JSON payload of the request.
+        """
+        request_kwargs = deepcopy(kwargs)
+
+        validate_request_params(
+            request_kwargs,
+            required_params={'fbc_fragment', 'from_index'},
+            optional_params={
+                'add_arches',
+                'binary_image',
+                'build_tags',
+                'overwrite_from_index',
+                'overwrite_from_index_token',
+            },
+        )
+
+        # Validate parent_bundle_image is correctly provided
+        fbc_fragment = request_kwargs.get('fbc_fragment')
+        if not isinstance(fbc_fragment, str):
+            raise ValidationError('The "fbc_fragment" must be a string')
+        request_kwargs['fbc_fragment'] = Image.get_or_create(pull_specification=fbc_fragment)
+
+        # cast to more wider type, see _from_json method
+        cls._from_json(
+            cast(RequestPayload, request_kwargs),
+            additional_optional_params=[
+                'bundles',
+                'fbc_fragment',
+                'from_index',
+                'organization',
+            ],
+        )
+
+        request = cls(**request_kwargs)
+        request.add_state('in_progress', 'The request was initiated')
+        return request
+
+    def to_json(self, verbose: Optional[bool] = True) -> FbcOperationRequestResponse:
+        """
+        Provide the JSON representation of a "fbc-operation" build request.
+
+        :param bool verbose: determines if the JSON output should be verbose
+        :return: a dictionary representing the JSON of the build request
+        :rtype: dict
+        """
+        # cast to result type, super-type returns Union
+        rv = cast(FbcOperationRequestResponse, super().to_json(verbose=verbose))
+        rv.update(self.get_common_index_image_json())  # type: ignore
+        rv['fbc_fragment'] = self.fbc_fragment.pull_specification
```

Comment:
removed_operators, organization, deprecation_list, bundles, bundle_mapping can be removed here

Reply:
Thank you - removed